### PR TITLE
St1k4

### PR DESCRIPTION
**Pull request by ghalym**
## Description
ST1K4 is a test stopper that is controlled by the kfe-motion PLC. changes made on kfe-motion to send the encoder value to the arbiter over the PMPS_PRE interface. https://github.com/pcdshub/lcls-plc-kfe-motion/pull/125
Now the arbiter evaluated the encoder value based on ranged recieved from @tongju12 and evaluated the veto.

Additional changes are a big fix for the FFO from the TXI K3 PLC.
Additional helper functions to remove certain requests from the arbitration manually, specially helpful with the K3 line not being fully commissioned but still is needed for special experiments run.

## Motivation and Context
TMO wants to use this device as a veto for the TMO PLCs.

## How Has This Been Tested?
Yes, communication between kfe-motion to arbiter and reporting the veto to the TMO PLCs all work ok.

## Where Has This Been Documented?
https://docs.google.com/document/d/196hyjN4b-y3hUMMy3MdFw4OegQlso_sLPnk2Q0doqRs/edit

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive comments
- [x] Test suite passes locally
- [x] Libraries are set to fixed versions and not ``Always Newest``
- [x] Code committed with ``pre-commit`` (alternatively ``pre-commit run --all-files``)

### CODE REVIEW

File: Arbiter/KFEArbiter/LineIDs.dbg
Comment (ZLLentz) [inline Arbiter/KFEArbiter/LineIDs.dbg:1]: This and all other .dbg files should be removed from the repository, we can do it in a separate PR if you like since it won't affect how the PLC runs at all.

File: Arbiter/KFEArbiter/POUs/CurrentBPUpdate.TcPOU [hunk 1]
```diff
@@ -159,35 +177,41 @@ fbEPICSLMachineMode(BP:=PMPS_GVL.stCurrentBeamParameters, fbMPS_MachineMode:=fbS
 st3k4Watcher(stCurrentBP:=PMPS_GVL.stCurrentBeamParameters);
 st1k2Watcher(stCurrentBP:=PMPS_GVL.stCurrentBeamParameters);
 st1k3Watcher(stCurrentBP:=PMPS_GVL.stCurrentBeamParameters);
+// ST4K4 Photon Terminator Veto
+st4k4Watcher(stCurrentBP:=PMPS_GVL.stCurrentBeamParameters);
```
Comment (ZLLentz) [inline Arbiter/KFEArbiter/POUs/CurrentBPUpdate.TcPOU:181]: This change isn't mentioned in the PR text anywhere. Was ST4K4 inactive as a stopper/veto device prior to this PR?

File: Arbiter/KFEArbiter/POUs/Fast_Faults_Evaluate.TcPOU [hunk 2]
```diff
@@ -33,21 +33,21 @@ END_VAR
...
 FFO3(
     io_fbFFHWO := g_FastFaultOutput2,
     i_xOK := i_xFastFaultOut_txi_mot,
     i_xAutoReset := TRUE);
-	
+
 FFO4(
     io_fbFFHWO := g_FastFaultOutput2,
-    i_xOK := i_xFastFaultOut_txi_mot,
+    i_xOK := i_xFastFaultOut_txi_optics,
```
Comment (ZLLentz) [inline Arbiter/KFEArbiter/POUs/Fast_Faults_Evaluate.TcPOU:50]: This change isn't mentioned in the PR text anywhere. What's this change for?

File: Arbiter/KFEArbiter/POUs/PMPS_Arbiter.TcPOU
Comment (ZLLentz) [inline Arbiter/KFEArbiter/POUs/PMPS_Arbiter.TcPOU:1]: Are the changes in this file intended to be included in the PR? It looks like these are debug helper functions. I think there is value in keeping them but they should also be in the PR description as documentation.

**Review by ZLLentz — COMMENTED** [review]
The code all looks good but I think there are undocumented changes in here that should be documented appropriately.

File: Arbiter/KFEArbiter/POUs/Fast_Faults_Evaluate.TcPOU (hunk 2 above)
Comment (ghalym) [reply Arbiter/KFEArbiter/POUs/Fast_Faults_Evaluate.TcPOU:50]: Apologies, this is a bug that i noticed and fixed with this PR. Will add it to the PR description.

File: Arbiter/KFEArbiter/LineIDs.dbg
Comment (ZLLentz) [reply Arbiter/KFEArbiter/LineIDs.dbg:1]: There is already an issue for this at #10 but we can continue to deprioritize it and work on other things. I won't bother putting it into a Jira.
